internal/storage/postgres/watering_plan/create.go

```diff
@@ -0,0 +1,87 @@
+package wateringplan
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	sqlc "github.com/green-ecolution/green-ecolution-backend/internal/storage/postgres/_sqlc"
+	"github.com/green-ecolution/green-ecolution-backend/internal/utils"
+)
+
+func defaultWateringPlan() *entities.WateringPlan {
+	return &entities.WateringPlan{
+		Date:               time.Time{},
+		Description:        "",
+		Distance:           utils.P(0.0),
+		TotalWaterRequired: utils.P(0.0),
+		Users:              make([]*entities.User, 0),
+		Treecluster:        make([]*entities.TreeCluster, 0),
+		Transporter:        nil,
+		Trailer:            nil,
+	}
+}
+
+func (w *WateringPlanRepository) Create(ctx context.Context, wFn ...entities.EntityFunc[entities.WateringPlan]) (*entities.WateringPlan, error) {
+	entity := defaultWateringPlan()
+	for _, fn := range wFn {
+		fn(entity)
+	}
+
+	if err := w.validateWateringPlan(entity); err != nil {
+		return nil, err
+	}
+
+	id, err := w.createEntity(ctx, entity)
+	if err != nil {
+		return nil, w.store.HandleError(err)
+	}
+
+	entity.ID = *id
+	return w.GetByID(ctx, *id)
+}
```

Comment:
Maybe we can already use the transaction here (not tested the suggestion)
```suggestion
func (w *WateringPlanRepository) Create(ctx context.Context, createFn func(*entities.WateringPlan) (bool, error)) (*entities.WateringPlan, error) {
	if createFn == nil {
		return nil, errors.New("createFn is nil")
	}
	
	var createdW *entities.WateringPlan
	err := w.store.WithTx(ctx, func(s *store.Store) err {
		oldStore := r.store
		defer func() {
			r.store = oldStore
		}()
		r.store = s
		
		entity := defaultWateringPlan()
		created, err := createFn(entity)
		if err != nil {
			return err
		}

		if !created {
			return nil
		}

		if err := w.validateWateringPlan(entity); err != nil {
			return nil, err
		}

		id, err := w.createEntity(ctx, entity)
		if err != nil {
			return nil, w.store.HandleError(err)
		}
		
		createdW, err = w.GetByID(ctx, *id)
		if err != nil {
			return err
		}
		
		return nil
	}
	return createdW, nil
}
```